pkg/kubectl/cmd/drain.go

```diff
@@ -656,3 +679,12 @@ func changed(desired bool) string {
 	}
 	return "uncordoned"
 }
+
+func (o *DrainOptions) sendEventWithMessage(desired bool) {
+	reason := "NodeUncordoned"
+	if desired {
+		reason = "NodeCordoned"
+	}
+
+	o.eventRecorder.Eventf(o.nodeInfo.Object, api.EventTypeNormal, reason, "Node %s: %s", o.nodeInfo.Name, o.Message)
```

Comment:
Why not just build your event and send it directly with a clientset?  The event recorder and broadcaster only provides value if you have multiple sinks or multiple correlated messages, otherwise its just one more thing to include that obscures what happens.

As an example of why this is problematic, consider the error case. Using  a client would let you explicitly ignore the failure if it happened and make it obvious to readers who don't know the event recorder very where.  As it is, I think this code glogs an info message out on failed sends.

The other consider is asynchronicity.  The event broadcaster uses some async channels before actually writing.  Now this works well for something like a controller, but for a client command it means you're racing to finish before the command completes.